# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/serviceworkers/ServiceWorkerOp.cpp
# Commit: cf972fe0d961
# Full Hash: cf972fe0d96164788f998f04008930474526e400
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2022-02-26 21:36:36
# Regressor Bug: 1754365
# File Overlap Count: 1
# Description:
#   Bug 1754365 - Create IPC for transferring NavigationPreload's PerformanceTimingData. r=dom-worker-reviewers,jesup
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D138251
# ==============================================================================

diff -r f8afd25bf41e -r cf972fe0d961 dom/serviceworkers/ServiceWorkerOp.cpp
--- a/dom/serviceworkers/ServiceWorkerOp.cpp	Sat Feb 26 08:40:25 2022 +0000
+++ b/dom/serviceworkers/ServiceWorkerOp.cpp	Sat Feb 26 08:40:26 2022 +0000
@@ -1682,10 +1682,16 @@
         ->Then(
             GetCurrentSerialEventTarget(), __func__,
             [self, globalObjectAsSupports = std::move(globalObjectAsSupports)](
-                SafeRefPtr<InternalResponse> aInternalResponse) {
-              self->mPreloadResponse->MaybeResolve(
-                  MakeRefPtr<Response>(globalObjectAsSupports,
-                                       std::move(aInternalResponse), nullptr));
+                FetchEventPreloadResponseArgs&& aArgs) {
+              SafeRefPtr<InternalResponse> preloadResponse;
+              IPCPerformanceTimingData timingData;
+              nsString initiatorType;
+              nsString entryName;
+              Tie(preloadResponse, timingData, initiatorType, entryName) =
+                  std::move(aArgs);
+
+              self->mPreloadResponse->MaybeResolve(MakeRefPtr<Response>(
+                  globalObjectAsSupports, std::move(preloadResponse), nullptr));
               self->mPreloadResponsePromiseRequestHolder.Complete();
             },
             [self](int) {